### ISSUE TITLE
Extend / Eject story with SST ION

### ISSUE THREAD

**Issue by amir-ziaei** [sst/ion#381]
Hello,

I've recently been exploring SST (ION engine) and find it to be a fantastic tool. However, as I consider integrating it into a work-related project, I have some concerns which I'm hoping to hear the team's response to.

## Extendability
Is there a plan in place for handling scenarios where some infra components are not yet supported by SST? Could you provide insights into how you expect the consumer to address such gaps? Are there any plans on your end to implement a plugin interface to extend support for additional components?

## Ejectablity
In the event that a user wishes to discontinue using SST, is there a means to access the underlying IoC that SST manages? How easy would that be?

---

Looking forward to hearing from the team!
Thanks!

**Comment by san4d** [comment]
Not part of the team but am using Ion and used v2.

All SST components are Pulumi components, so the "lock in" is with Pulumi. In my project, I'm using both SST components and Pulumi components together seamlessly. When there isn't a prebuilt component, I drop down and write my own Pulumi component. 

If I wanted to move off of SST, I would reimplement each of the crafted SST components myself as Pulumi components (and have to figure out SDLC things, like a replacement for live lambda). If I wanted to move off of Pulumi, that would be a larger task. You can always look at your cloud provider to see the exact resources created. I recommend doing this anyhow.

https://ion.sst.dev/docs/components/

**Comment by jayair** [comment]
Yeah @san4d basically covered it all, I would add that the extend path when Ion does not support it is:

- Try, the `transform`: https://ion.sst.dev/docs/components/#transform
- Then try Pulumi directly
- If not supported by Pulumi but there's a TF provider, then there are steps to generate a Pulumi component as well, we do this ourselves in some cases: https://github.com/sst/pulumi-supabase

**Comment by iDVB** [comment]
If I'm understanding this correctly I think @thdxr mentioned that ion extending issues is something that is being fixed?
https://discord.com/channels/983865673656705025/1177071497974648952/1257351758984904758

Do we have a public roadmap and where on it is this work? I ask because it may make more sense for us to convert from sstv2 -> ion after that point. So we will have a path to convert all our custom CDK/SST constructs that in many cases use or extend SST constructs.

**Comment by jayair** [comment]
@iDVB I might be wrong but I think yours is more about creating components internally that your team can reuse right?
